src/ooxml/java/org/apache/poi/xssf/streaming/IRowGenerator.java

```diff
@@ -0,0 +1,35 @@
+/* ====================================================================
+   Licensed to the Apache Software Foundation (ASF) under one or more
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+==================================================================== */
+
+package org.apache.poi.xssf.streaming;
+
+import org.apache.poi.util.Beta;
+
+/**
+ * IRowGenerator for Super-streaming XSSF sheets
+ */
+@Beta
+public interface IRowGenerator {
+    
+    /**
+     * Generate and add rows to the sheet
+     *
+     * @param sheet the sheet
+     * @throws Exception the exception
+     */
+    void generateRows(SXSSFSheet sheet) throws Exception;
```

Comment:
I don't quite understand this - why not just add rows using Sheet#createRow?

https://github.com/apache/poi/pull/141/files#diff-ea5bbd436cd5c48efd0716b8d6f0e6ebR155 (highlight line in yellow) - seems a very verbose way to create a row

Reply:
@pjfanning `SuperSXSSFSheet` does not initialize its `_writer` until `workbook.write(out);` is already running.  Row and cell generation happens in `SXSSFWorkbook#injectData()` and `#copyStreamAndInjectWorksheet()`. This is what allows for direct streaming to the client.

Reply:
Not 100% true: you can call `#createRow` as long as you stay within `randomAccessWindowSize` of SXSSFSheet.